sympy/ntheory/residue_ntheory.py

```diff
@@ -773,6 +777,8 @@ def nthroot_mod(a, n, p, all_roots=False):
     # see Hackman "Elementary Number Theory" (2009), page 76
     if not is_nthpow_residue(a, n, p):
         return None
+    if isprime(p) is False:
+        raise NotImplementedError("Not implemented for composite p")
     if primitive_root(p) is None:
         raise NotImplementedError("Not Implemented for m without primitive root")
```

Comment:
If `p` is prime, then a primitive root exists. So these two lines can be removed.

Reply:
But we have to add a primality test for the modulo right. Without it we were getting wrong answers.

Reply:
The primality test on line 776 is currently necessary, but if `p` is prime, then a primitive root exists and the test on line 778 will always fail. So the two lines 778-779 can be removed.